src/menelaus/ensemble.py

```diff
@@ -42,26 +149,9 @@ class Ensemble:
 
     def __init__(self, detectors: dict, evaluator: str, columns: dict = None):
         self.detectors = detectors.copy()
-        self.evaluator = evaluators[evaluator]
+        self.evaluator = EVALUATORS[evaluator]
```

Comment:
Supposing I wanted to pass in e.g. `eval_minimum_approval(2)` as opposed to one of the string options, I could `import EVALUATORS` and then add a key. Should we make a note for a bit later to support accepting functions directly?

Reply:
@tms-bananaquit I think the recommended approach would just be to import the evaluator function, and we could get away with not doing anything more:

```python
from menelaus.ensemble import eval_confirmed_approval

evaluator = eval_confirmed_approval(approvals_needed=1, confirmations_needed=3)
ensemble = Ensemble(detectors, evaluator)
```

Thoughts?

Reply:
With the current implementation, this fails out in L152, where it tries `self.evaluator = EVALUATORS[evaluator]`. I think this needs a check on the type, at least, to allow passing functions directly.

Reply:
Oh, oops. Fixing now

Reply:
The change I arrived at is just making the `evaluator` parameter a function all the time. So either:

```python
Ensemble(detectors, evaluator_fn)
Ensemble(detectors, EVALUATORS['simple-majority']
```

I'd like to avoid what I feel would be discouraged type-checking (string/function) for the parameter, but that comes at the cost of making users add this ugly piece into the constructor. Let me know what you think @tms-bananaquit

Reply:
Would it gain us anything to use a defaultdict here, too, and assign the evaluator directly if there's no entry for it in EVALUATORS? It's kind of sneaking in the type-checking, I guess, except failure would look like an error of "I can't execute whatever garbage you handed me as a function!". I'm _assuming_ checking membership is more performant than checking type?

Reply:
I can't figure out how to include functions in the table. A `defaultdict` would let us return some default value for non-existent keys, but I'm not sure how to meaningfully correspond a "bad" key passed to the table, to e.g. `eval_confirmed_approval( some fixed parameterization)`.

```python
self.evaluator = EVALUATORS['simple-majority']      # this returns a function
self.evaluator = EVALUATORS['bad-value']            # how do we go from bad-value to one of our functions
```

The avoidance of type-checking is less of a performance thing and more of an adherence to duck-typing. I'll see if I can find something more concrete regarding this.

Reply:
Something like this?
```
#returns False if the key is missing
EVALUATORS = defaultdict(lambda: False)
...
#sets the evaluator to the input, otherwise tries to match string
self.evaluator = key if not EVALUATORS[key] else EVALUATORS[key] 
```
But yeah, I get the concern. Trying to come up with something to avoid having to pass `EVALUATORS['blah']` and also always having to `from menelaus.ensemble import EVALUATORS` every time.

Reply:
Your original `if string, lookup and if function, execute` idea might be the clearest for developers and users alike. Opting for that for now @tms-bananaquit and leaving this pseudocode in case of future issues for evaluators.

```python
# essentially a closure, but may have some nice properties to let users query evaluator 
# AND may allow for the down-the-road ability to 'wait' for burn-in syncing
class ConfirmedApprovalEvaluator():
    def __init__(self, approvals_needed=2):
        self.approvals_needed = approvals_needed

    def __call__(self, detectors):
        # do voting

# somewhere in Ensemble code...
self.evaluator(detectors)

e = Ensemble(detectors, ConfirmedApprovalEvaluator())
```

Reply:
That makes sense. Reverting the last commit, since that version is closer to this solution than the others.